src/fmu/tools/ensembles/ensemble_well_props.py

```diff
@@ -0,0 +1,616 @@
+      # is possible and the result will be a "AND" combination
+      cumulative_lengths:
+        # output will be one file for reals, and one file for statistics e.g.
+        # cumlen.csv and cumlen_summary.csv
+        fileroot: cumlen
+        criteria:
+          Facies:
+            codes: [1, 3, 4] # several numbers possible
+          PHIT:
+            interval: [0.22, 0.4]  # only two numbers
+          PERM:
+            interval: [1000, 100000]  # only two numbers
+      # the fields below gives an opportunity to study more details of the calculations
+      keep_intermediate_logs: false
+      show_in_terminal: true
+
+    """
+
+    example = example.split("\n")[1:]
+
+    with open("example.yml", "w") as stream:
+        for line in example:
+            line = line.replace(" " * 4, "", 1)
+            stream.write(f"{line}\n")
+    print("See example.yml, and use this as template for your case!")
+
+
+def parse_config(configfile):
+    """Parse YAML file."""
+    try:
+        with open(configfile, "r", encoding="utf8") as stream:
+            config = yaml.safe_load(stream)
+    except IOError:
+        print(f"Could not read file: {configfile}")
+        raise
+
+    return config
+
+
+@dataclass
+class PropsData:
+    """Holding property data and criteria"""
+
+    name: str
+    filestub: str
+    discrete: bool = False
+    codes: Optional[list] = None
+    interval: Optional[list] = None
+
+
+@dataclass
+class ConfigData:
+    """Other class data than config are evaluated from config in post_init."""
+
+    config: dict
+    root: Optional[Path] = None
+    reals: Optional[list] = None
+    itera: Optional[str] = None
+
+    wellfile: Optional[Path] = None
+    welldelta: int = 2
+
+    gridfile: Optional[Path] = None
+    gridreuse: bool = False
+
+    proplist: Optional[List[PropsData]] = None
+
+    report_avg_file: Optional[str] = None
+    report_cum_file: Optional[str] = None
+    report_keep_intermediate: bool = False
+    report_show_in_terminal: bool = True
+
+    def __post_init__(self):
+        """Derive input data from config."""
+
+        self.root = Path(self.config["ensemble"]["root"])
+
+        if "range" in self.config["ensemble"]["realizations"]:
+            start, stop = self.config["ensemble"]["realizations"]["range"].split("-")
+            self.reals = list(range(int(start), int(stop) + 1))
+        elif "entries" in self.config["ensemble"]["realizations"]:
+            self.reals = self.config["ensemble"]["realizations"]["entries"]
+        else:
+            raise ValueError("For 'realizations': 'range' or 'entries' is missing")
+
+        self.itera = self.config["ensemble"]["iteration"]
+
+        self.wellfile = self.config["well"]["file"]
+        self.lognames = self.config["well"].get("lognames", "all")
+        self.mdlog = self.config["well"].get("mdlog", "MDepth")
+        self.mdranges = self.config["well"]["mdranges"]
```

Comment:
Add a validation of mdranges intervals to ensure no intervals are overlapping and hence impacting the total cumulative length